models/news-model.js

```diff
@@ -53,19 +56,20 @@ exports.insertNewComment = (newComment, article_id) => {
     .query(checkingUserQuery, [username])
     .then((authorsArray) => {
       if (authorsArray.rows.length === 0) {
-        return Promise.reject();
+        return Promise.reject({ status: 404, msg: "Error: 404 not found" });
       }
-      return authorsArray.rows[0];
+      return Promise.all(authorsArray.rows);
     })
-    .then((user) => {
-      const { username } = user;
+    .then((authors) => {
+      const { username } = authors[0];
```

Comment:
You have access to the username from your newComment variable, so you don't need to take it from your authors array. Totally works though!